### DISCUSSION TITLE
koyeb怎么连接数据库

### DISCUSSION THREAD

**Discussion by ttt789-ppo** [AlistGo/alist#931] [Q&A]
我是用koyeb搭建的，这种怎么连接数据库，看网上说用docker拉取mysql镜像，但是没用

**Comment by zxmacc** [comment]


**Reply by SuCan127** [reply]
你好，我按照你的教程搭了一遍，https://www.zxma.top/posts/438365eb/#%E8%AE%BE%E7%BD%AE%E7%8E%AF%E5%A2%83%E5%8F%98%E9%87%8F,请问你是怎么实现webdav分享的，我怎么也成功不了

**Reply by zxmacc** [reply]
抱歉，不常看，webdav的话，在后台管理的账号一栏，添加就行了，我截图给你看一下
![image](https://user-images.githubusercontent.com/71143103/185175705-d43b4327-35f9-4c7a-8d53-ec9bade1672c.png)

**Comment by zxmacc** [comment]


**Reply by zxmacc** [reply]
确实，体验到了，不过能用就行，我也没别的好方法了

**Reply by zxmacc** [reply]
哦哦，好滴，我等等去试试，谢谢啦OvO

**Reply by zxmacc** [reply]
> render+elephantsql的Postgre数据库访问速度就很不错

大佬~，render的一键部署好像有点问题，我把koyeb上成功的变量填进去还是显示An error has occurred.，大佬能提供点帮助吗

**Reply by zxmacc** [reply]
话说，Postgre数据库的连接方式和mysql的一样吗，还是要和heroku部署上一样，就填DATABASE_URL的变量QwQ，我在用replit去模拟，目前是死活连不上，数据就是不写进去

**Reply by zxmacc** [reply]
> 目前在我看来 replit koyeb render里就render裸连速度最快 elephantsql的Postgre数据库能选服务地 选跟alist服务地接近的 连接会快不少

好的，感谢大佬，已经部署成功了，并且裸连速度也很快OwO

**Reply by zxmacc** [reply]
> 嗷 对了 用uptimerobot监视网站 保持唤醒 render 15分钟没访问会休眠 休眠重新唤醒还是有点慢的

感受到了，唤醒大概要半分钟多一点QwQ，感谢大佬的建议，这就去弄~

**Comment by zxmacc** [comment]
好吧，解决了，淦，我悟了
这样配置就可以了
{
  "DB_TYPE": "mysql",
  "DB_HOST": "db4free.net",
  "DB_PORT": "3306",
  "DB_USER": "",
  "DB_PASS": "",
  "DB_NAME": "",
  "DB_TABLE_PREFIX": "alist_",
  "DB_SLL_MODE": "false"
}

**Reply by ttt789-ppo** [reply]
> 好吧，解决了，淦，我悟了 这样配置就可以了 { "DB_TYPE": "mysql", "DB_HOST": "db4free.net", "DB_PORT": "3306", "DB_USER": "", "DB_PASS": "", "DB_NAME": "", "DB_TABLE_PREFIX": "alist_", "DB_SLL_MODE": "false" }

这种没用的，因为你添加的所有文件包括配置文件都会随着koyeb容器重启而重置，除非有环境变量，koyeb能配置环境变量，但需要拉取的镜像本身有环境变量你才能配置，不然你在容器所做的所有修改都会随着容器重启而重置

**Reply by zxmacc** [reply]
是环境变量，我图方便写了一下

**Reply by wintopic** [reply]
请问CDN应该填写什么

**Reply by zuoan286** [reply]
我也koyeb上部署的，重启后资料就没了，麻烦问一下容器的环境变量怎么写的
